src/core/algorithms/dc/parser/dc_parser.cpp

```diff
@@ -0,0 +1,168 @@
+        throw std::invalid_argument("Missing parenthesis");
+
+    dc_string_.erase(dc_string_.begin());
+    dc_string_.erase(std::prev(dc_string_.end()));
+
+    if (dc_string_.empty()) throw std::invalid_argument("Empty DC is not allowed");
+
+    std::vector<Predicate> res;
+    while (has_next_predicate_) {
+        Predicate pred = GetNextPredicate();
+        res.push_back(pred);
+    }
+
+    return {res};
+}
+
+Predicate DCParser::GetNextPredicate() {
+    size_t ind = dc_string_.find(sep_, cur_);
+    if (ind == std::string::npos) has_next_predicate_ = false;
+
+    std::string str_pred = dc_string_.substr(cur_, ind - cur_);
+    cur_ = ind + sep_.size();
+
+    return ConvertToPredicate(std::move(str_pred));
+}
+
+Predicate DCParser::ConvertToPredicate(std::string pred) {
+    size_t ind;
+    std::string str_operator;
+    for (std::string const& op : str_operators_) {
+        std::string spaced_op = ' ' + op + ' ';
+        ind = pred.find(spaced_op);
+        if (ind != std::string::npos) {
+            str_operator = op;
+            break;
+        }
+    }
+
+    if (ind == std::string::npos) throw std::invalid_argument("Missing permissible operator");
+
+    size_t right_start = ind + str_operator.size() + 1;
+    auto str_left_operand = std::string(pred, 0, ind);
+    auto str_right_operand = std::string(pred, right_start, pred.size() - right_start + 1);
+
+    boost::trim(str_left_operand);
+    boost::trim(str_right_operand);
+
+    auto [left_operand, right_operand] = GetOperands(str_left_operand, str_right_operand);
+
+    return {std::move(str_operator), std::move(left_operand), std::move(right_operand)};
+}
+
+std::pair<ColumnOperand, ColumnOperand> DCParser::GetOperands(std::string str_left_op,
+                                                              std::string str_right_op) {
+    std::unique_ptr<ColumnOperand> left_op, right_op;
+
+    if (IsVarOperand(str_left_op)) {
+        RemoveEscaping(str_left_op);
+        ColumnOperand operand = ConvertToVariableOperand(str_left_op);
+        left_op = std::make_unique<dc::ColumnOperand>(operand);
+    }
+
+    if (IsVarOperand(str_right_op)) {
+        RemoveEscaping(str_right_op);
+        ColumnOperand operand = ConvertToVariableOperand(str_right_op);
+        right_op = std::make_unique<dc::ColumnOperand>(operand);
+    }
+
+    if (!left_op and !right_op)
+        throw std::invalid_argument("Pure constant predicate is not allowed");
+
+    if (!left_op) {
+        left_op = GetOperand(*right_op.get(), str_left_op);
+    } else if (right_op == nullptr) {
+        right_op = GetOperand(*left_op.get(), str_right_op);
+    }
+
+    return {std::move(*left_op), std::move(*right_op)};
+}
+
+void DCParser::RemoveEscaping(std::string& str) const {
+    size_t ind = str.find_first_of('\\');
+    if (ind != std::string::npos) str.erase(ind, 1);
+}
+
+bool DCParser::IsVarOperand(std::string op) const {
+    return op.find("t.") == 0 or op.find("s.") == 0;
```

Comment:
```suggestion
    return op.starts_with("t.") || op.starts_with("s.");
```
I love C++20, didn't know we can do this :)